maat-court-data-api/src/test/java/gov/uk/courtdata/integration/link/service/CreateLinkListenerIntegrationTest.java

```diff
@@ -85,7 +87,11 @@ public void givenSaveAndLinkModel_whenSaveAndImplIsInvoked_thenLinkEstablished()
         String saveAndLinkMessage = testModelDataBuilder.getSaveAndLinkString();
 
         //when
-        createLinkListener.receive(saveAndLinkMessage, "2");
+        Map<String, Object> header = new HashMap<>();
+        header.put("MessageId", "AIDAIU3GACVJITZULQ2RQ");
+        MessageHeaders headers = new MessageHeaders(header);
+        //when
+        createLinkListener.receive(saveAndLinkMessage, headers);
```

Comment:
I would argue this is not really an integration test as it's not testing SQSListener reading from the message queue.

A local test message queue can be set up and populated with the exact message data to test that this SQSListener is doing as we expect.

We can investigate doing that at another time though - not in the scope of this ticket.

Reply:
Yes it would be nice to have a full integration test for this one - @lkm0287 can you add in a spike to the backlog to explore options please?